metrics-facade-base/src/main/java/com/ringcentral/platform/metrics/instrument/executors/AbstractMonitoredExecutorServiceBuilder.java

```diff
@@ -0,0 +1,176 @@
+
+    private final ES parent;
+    private final MetricRegistry registry;
+    private String name;
+    private boolean nameAsLabel = true;
+    private MetricName metricNamePrefix = DEFAULT_METRIC_NAME_PREFIX;
+    private LabelValues prefixLabelValues;
+    private LabelValues additionalLabelValues = noLabelValues();
+
+    protected static final AtomicLong orderNum = new AtomicLong();
+
+    /**
+     * Returns the next order number as a string (e.g., "1", "2", ...).
+     * This is used to generate a metric name when no explicit name is provided.
+     *
+     * @return the next order number as a string
+     */
+    protected static String nextOrderNumAsString() {
+        return Long.toString(orderNum.incrementAndGet());
+    }
+
+    /**
+     * @param parent   the parent {@link ExecutorService} that is being monitored
+     * @param registry the {@link MetricRegistry} used to register metrics
+     */
+    public AbstractMonitoredExecutorServiceBuilder(@Nonnull ES parent, @Nonnull MetricRegistry registry) {
+        this.parent = parent;
+        this.registry = registry;
+    }
+
+    /**
+     * Sets the name of the executor service.
+     * The name will either be included as part of the metric names (when {@code nameAsLabel(false)} is used),
+     * or as the value of the "name" label (the default behavior with {@code nameAsLabel(true)}).
+     *
+     * @param name the name to assign to the executor service
+     * @return this builder
+     */
+    public B name(String name) {
+        this.name = name;
+        return builder();
+    }
+
+    /**
+     * Configures whether the name of the executor service should be used as the value of the "name" label
+     * or included directly as part of the metric names.
+     *
+     * @param nameAsLabel {@code true} to use the name as a label, {@code false} to include it as part of the metric names
+     * @return this builder
+     */
+    public B nameAsLabel(boolean nameAsLabel) {
+        this.nameAsLabel = nameAsLabel;
+        return builder();
+    }
+
+    /**
+     * Sets the metric name prefix.
+     *
+     * @param metricNamePrefix the prefix to use for the metric names
+     * @return this builder
+     * @throws IllegalArgumentException if {@code metricNamePrefix} is empty
+     */
+    public B metricNamePrefix(@Nonnull MetricName metricNamePrefix) {
+        checkArgument(!metricNamePrefix.isEmpty(), "metricNamePrefix is empty");
+        this.metricNamePrefix = metricNamePrefix;
+        return builder();
+    }
+
+    /**
+     * Sets the label values to prefix the metrics with.
+     *
+     * @param prefixLabelValues the label values to prefix the metrics with
+     * @return this builder
+     */
+    public B prefixLabelValues(LabelValues prefixLabelValues) {
+        this.prefixLabelValues = prefixLabelValues;
+        return builder();
+    }
+
+    /**
+     * Sets additional label values to be applied to the metrics.
```

Comment:
There is a slight difference in semantics: prefix label values take precedence over all modifications made in AbstractMetricRegistry. See the usages of AbstractMetricRegistry.replacePrefixLabelValues(). While this behavior is debatable, it reflects the current implementation, and we should allow users to set these labels separately. We could suggest that users inject additional labels into the prefix ones using this builder, but that would contradict a common pattern in the library, where prefix labels are defined solely through post-configuration without using PrefixLabelValuesMetricKey. If we were to allow this, users would have to manage both prefix and additional labels in the builder, which would be inconvenient. They would first define the prefix labels once during post-configuration, but for this specific builder, they would then need to combine them manually

Reply:
Could we add link to the description of this method `AbstractMetricRegistry#replacePrefixLabelValues` or maybe some docs with description of this behaviour? 
I'm afraid that it could be a bit complicated for lib's users to understand this difference without any context.

Reply:
The description of these features deserves dedicated client documentation and JavaDocs. I've created a follow-up task: https://github.com/ringcentral/metrics-facade/issues/67